src/conditions/base/contract.ts

```diff
@@ -6,31 +7,54 @@ import { RpcCondition, rpcConditionRecord } from './rpc';
 
 export const STANDARD_CONTRACT_TYPES = ['ERC20', 'ERC721'];
 
-const functionAbiVariable = Joi.object({
-  internalType: Joi.string(), // TODO is this needed?
-  name: Joi.string().required(),
-  type: Joi.string().required(),
-});
-
 const functionAbiSchema = Joi.object({
   name: Joi.string().required(),
   type: Joi.string().valid('function').required(),
-  inputs: Joi.array().items(functionAbiVariable),
-  outputs: Joi.array().items(functionAbiVariable),
-  // TODO: Should we restrict this to 'view'?
-  stateMutability: Joi.string(),
+  inputs: Joi.array(),
+  outputs: Joi.array(),
+  stateMutability: Joi.string().valid('view', 'pure').required(),
 }).custom((functionAbi, helper) => {
+  // Is `functionABI` a valid function fragment?
+  let asInterface;
+  try {
+    asInterface = new ethers.utils.Interface([functionAbi]);
+  } catch (e: unknown) {
+    const { message } = e as Error;
+    return helper.message({
+      custom: message,
+    });
+  }
+
+  if (!asInterface.functions) {
+    return helper.message({
+      custom: '"functionAbi" is missing a function fragment',
+    });
+  }
+
+  if (Object.values(asInterface.functions).length !== 1) {
+    return helper.message({
+      custom: '"functionAbi" must contain exactly one function fragment',
+    });
+  }
+
+  // Now we just need to validate against the parent schema
   // Validate method name
   const method = helper.state.ancestors[0].method;
-  if (functionAbi.name !== method) {
+  const abiMethodName = Object.keys(asInterface.functions).find((name) =>
+    name.startsWith(`${method}(`)
+  );
+  const functionFragment = abiMethodName
+    ? asInterface.functions[abiMethodName]
+    : null;
```

Comment:
Could this be reduced to the following (https://docs.ethers.org/v5/api/utils/abi/interface/#Interface--fragments)?

```suggestion
  const functionFragment = asInterface.getFunction(method);
```

It seems to possible throw an `ArgumentError` - do we care?

Reply:
I think it makes sense given our conversation about the ambiguity of functions. Fixed.